pkg/repositories/transformers/task_execution.go

```diff
@@ -152,6 +157,47 @@ func mergeLogs(existing, latest []*core.TaskLog) []*core.TaskLog {
 	return logs
 }
 
+func mergeCustom(existing, latest *_struct.Struct) (*_struct.Struct, error) {
+	if existing == nil {
+		return latest, nil
+	}
+	if latest == nil {
+		return existing, nil
+	}
+
+	// To merge latest into existing we first create a patch object that consists of applying changes from latest to
+	// an empty struct. Then we apply this patch to existing so that the values changed in latest take precedence but
+	// barring conflicts/overwrites the values in existing stay the same.
+	var empty _struct.Struct
```

Comment:
do you have to create this everytime? Can you not just create it once and re-use as a constant? Reading the docs makes me think its possible. This would reduce garbage

Reply:
Also have you looked at - https://pkg.go.dev/k8s.io/apimachinery/pkg/util/strategicpatch#CreateTwoWayMergePatch

Reply:
@kumare3 done

Reply:
@kumare3 re the link, this doesn't quite do what we need

> It will return a patch that yields the modified document when applied to the original document